src/sage/rings/fraction_field.py

```diff
@@ -653,19 +677,84 @@ def _element_constructor_(self, x, y=None, coerce=True):
+            Power Series Ring in x over Rational Field
+            sage: F(f)
+            doctest:warning...
+            DeprecationWarning: Conversion from power series to rational function field is deprecated, use .truncate() instead
+            See https://github.com/sagemath/sage/issues/39485 for details.
+            -x^19 + x^18 - x^17 + x^16 - x^15 + x^14 - x^13 + x^12 - x^11 + x^10 - x^9 + x^8 - x^7 + x^6 - x^5 + x^4 - x^3 + x^2 - x + 1
+
+        Conversion from Laurent series to rational function field gives an approximation::
+
+            sage: F.<x> = Frac(QQ['x'])
+            sage: R.<x> = QQ[[]]
+            sage: f = Frac(R)(1/(x+1))
+            sage: f.parent()
+            Laurent Series Ring in x over Rational Field
+            sage: F(f)
+            1/(x + 1)
+            sage: f = f.truncate(20); f  # infinite precision
+            1 - x + x^2 - x^3 + x^4 - x^5 + x^6 - x^7 + x^8 - x^9 + x^10 - x^11 + x^12 - x^13 + x^14 - x^15 + x^16 - x^17 + x^18 - x^19
+            sage: f.parent()
+            Laurent Series Ring in x over Rational Field
+            sage: F(f)
+            -x^19 + x^18 - x^17 + x^16 - x^15 + x^14 - x^13 + x^12 - x^11 + x^10 - x^9 + x^8 - x^7 + x^6 - x^5 + x^4 - x^3 + x^2 - x + 1
+            sage: f = 1/(x*(x+1))
+            sage: f.parent()
+            Laurent Series Ring in x over Rational Field
+            sage: F(f)
+            1/(x^2 + x)
+
+        ::
+
+            sage: K.<x> = FunctionField(QQ)
+            sage: R.<x> = QQ[[]]
+            sage: f = 1/(x+1)
+            sage: K(f)
+            doctest:warning...
+            DeprecationWarning: Conversion from power series to rational function field is deprecated, use .truncate() instead
+            See https://github.com/sagemath/sage/issues/39485 for details.
+            -x^19 + x^18 - x^17 + x^16 - x^15 + x^14 - x^13 + x^12 - x^11 + x^10 - x^9 + x^8 - x^7 + x^6 - x^5 + x^4 - x^3 + x^2 - x + 1
+            sage: f = Frac(R)(1/(x+1))
+            sage: K(f)
+            1/(x + 1)
+            sage: f = 1/(x*(x+1))
+            sage: K(f)
+            1/(x^2 + x)
         """
         if isinstance(x, (list, tuple)) and len(x) == 1:
             x = x[0]
         if y is None:
             if parent(x) is self:
                 return x
+            from sage.rings.polynomial.polynomial_ring import PolynomialRing_generic
+            if isinstance(self.ring(), PolynomialRing_generic):
+                from sage.rings.power_series_ring_element import PowerSeries
+                from sage.rings.laurent_series_ring_element import LaurentSeries
+                if isinstance(x, PowerSeries):
+                    from sage.misc.superseded import deprecation
+                    deprecation(
+                        39485,
+                        "Conversion from power series to rational function field is deprecated, use .truncate() instead",
+                    )
+                if isinstance(x, LaurentSeries):
```

Comment:
```suggestion
                if isinstance(x, PowerSeries):
                    from sage.misc.superseded import deprecation
                    deprecation(
                        39485,
                        "Conversion from power series to rational function field is deprecated, use .truncate() instead",
                    )
                    x = x.laurent_series()
                if isinstance(x, LaurentSeries):
```
I think power series should be handled in the same way as Laurent series. This isn't the most efficient way to do this, but it makes the code maintenance much easier.

Reply:
I agree (that's the whole point of introducing the deprecation), but is it fine if the behavior is just changed without a deprecation period?

Reply:
Hmm...I'm of the opinion as long as you tell people how to recover the old behavior, then it doesn't need to keep the old behavior.

Actually, looking more closely at the message in the current PR is incorrect. The conversion will still be there, but it will be done by a different method. So we probably should change that.